pkg/cmd/roachtest/registry/filter.go

```diff
@@ -60,9 +96,348 @@ func NewTestFilter(filter []string, runSkipped bool) *TestFilter {
+		nameOnlyFilter := noFilter
+		nameOnlyFilter.Name = filter.Name
+		if len(nameOnlyFilter.Filter(tests)) == 0 {
+			return nil, NoSuchName
+		}
+	}
+
+	// Check potential problems related to the suite.
+	if filter.Suite != "" {
+		// 3. Is the suite incorrect?
+		// We check if no tests match the suite, in which case the suite is the problem.
+		suiteOnlyFilter := noFilter
+		suiteOnlyFilter.Suite = filter.Suite
+		if len(suiteOnlyFilter.Filter(tests)) == 0 {
+			return nil, NoSuchSuite
+		}
+
+		// 4. Is the suite+regexp incorrect?
+		// We check if no tests match the suite AND the regexp.
+		suiteAndNameFilter := suiteOnlyFilter
+		suiteAndNameFilter.Name = filter.Name
+		if len(suiteAndNameFilter.Filter(tests)) == 0 {
+			return nil, NoTestsWithNameAndSuite
+		}
+	}
+
+	// Check potential problems related to the owner.
+	if filter.Owner != "" {
+		// 5. Is the owner incorrect?
+		// We check if no tests match the owner, in which case the owner is the problem.
+		ownerOnlyFilter := noFilter
+		ownerOnlyFilter.Owner = filter.Owner
+		if len(ownerOnlyFilter.Filter(tests)) == 0 {
+			return nil, NoSuchOwner
+		}
+		// 6. Is the owner+regexp incorrect?
+		// We check if no tests match the owner AND the regexp.
+		ownerAndNameFilter := ownerOnlyFilter
+		ownerAndNameFilter.Name = filter.Name
+		if len(ownerAndNameFilter.Filter(tests)) == 0 {
+			return nil, NoTestsWithNameAndOwner
+		}
+	}
+
+	// We failed to produce a useful message. It's an uncommon combination of
+	// criteria that leads to no tests matching.
+	return nil, NoHint
+}
+
+// NoMatchesHintString returns a user-friendly string describing the hint.
+func (filter *TestFilter) NoMatchesHintString(h NoMatchesHint) string {
+	noun := filter.noun()
+	switch h {
+	case WrongCloud:
+		return fmt.Sprintf(
+			"no %s match criteria; some %s match but are not compatible with %s",
+			noun, noun, filter.cloudStr(),
+		)
+	case NoSuchName:
+		return fmt.Sprintf("no %s match regexp %q", noun, filter.Name)
+	case NoSuchSuite:
+		return fmt.Sprintf("no %s in suite %q", noun, filter.Suite)
+	case NoTestsWithNameAndSuite:
+		return fmt.Sprintf("no %s in suite %q match regexp %q", noun, filter.Suite, filter.Name)
+	case NoSuchOwner:
+		return fmt.Sprintf("no %s with owner %q", noun, filter.Owner)
+	case NoTestsWithNameAndOwner:
+		return fmt.Sprintf("no %s with owner %q match regexp %q", noun, filter.Owner, filter.Name)
+	default:
+		return fmt.Sprintf("no %s match criteria", noun)
+	}
+}
+
+// Describe returns a multi-line string that describes the filter in a user-friendly way.
+// A verb like "Listing " or "Running" can be prepended to the first line. A
+// period can be prepended to the last line.
+//
+// Examples:
+// 1. all tests
+// 2. all benchmarks matching regexp "foo"
+// 3. all tests compatible with GCE
+// 4. tests which:
+//   - are compatible with GCE, and
+//   - have owner "foo", and
+//   - match regexp "bar"
+func (filter *TestFilter) Describe() []string {
```

Comment:
This is nice, but I think just printing the `TestFilter` struct would suffice, (or pretty printing). 

Seeing something like 
```
TestFilter { Cloud: GCE, Suite: Nightly }
```
is easy to parse.